connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java

```diff
@@ -80,7 +80,7 @@ public CorrelationResult<?> correlate(
           boundaryEventCorrelationPoint.messageName(),
           boundaryEventCorrelationPoint.correlationKeyExpression(),
           variables,
-          boundaryEventCorrelationPoint.messageIdExpression());
+          extractMessageId(boundaryEventCorrelationPoint.messageIdExpression(), variables));
```

Comment:
This would ignore the `messageId` value passed as a method parameter.

Reply:
Reworked completely